R/dependencies.R

```diff
@@ -0,0 +1,112 @@
+##' @title Get the dependencies for a given report from the orderly_run.yml
+##' 
+##' @param report the name of the report
+##' @param id the id of the report, if omitted, use the id of the latest report
+##' @param config_path Path to the orderly configuration
+##' @param draft If true look in the draft directory otehrwise look in archive
+##'
+##' @export
+get_dependencies <- function(name, id = NULL, config_path = NULL, draft = FALSE) {
+  if (is.null(id))
+    id <- orderly:::orderly_latest(name, config = config_path, draft = draft)
+  
+  if (is.null(config_path))
+    config_path <- "."
+
+  if (draft)
+    rds_path <- file.path(config_path, "draft", name, id, "orderly_run.rds")
+  else
+    rds_path <- file.path(config_path, "archive", name, id, "orderly_run.rds")
+
+  if (!file.exists(rds_path)) {
+    warning(sprintf(
+      "The report %s:%s does not exist or does not have orderly_run.rds",
+      name, id))
+    return(NULL)
+  }
+  run_data <- readRDS(rds_path)
+  return(run_data$meta$depends)
+}
+
+##' @title Generate a list of dependencies for a given report
+##'
+##' @param report the name of the report
+##' @param id the id of the report, if omitted, use the id of the latest report
+##' @param config_path Path to the orderly configuration
+##' @param draft If true look in the draft directory otehrwise look in archive
+##' @param depth only used internally, to prevent infinite recursion and make 
+##'              the output log look nice
+##' @param dep_reports the list of reports, needs to passed as an argument
+##'                          (and returned)
+##'
+##' @export
+what_depends_on <- function(report, id = "latest", config_path = NULL,
+                            draft = FALSE, depth = 0,
+                            dep_reports = list()) {
+  # this is potentially needed to trap circular dependencies
+  # TODO - We could be clever here and check if this report already appears in
+  # dep_reports
+  if (depth > 10) {
+    orderly::orderly_log("depends",
```

Comment:
don't issue a message that is a warning!  This should probably be an error, but it should also be disallowed in the current database building